lib/Sema/ConstraintSystem.cpp

```diff
@@ -2352,25 +2353,50 @@ static bool diagnoseConflictingGenericArguments(ConstraintSystem &cs,
                      [](const auto &arg) { return arg->hasPlaceholder(); }))
       continue;
 
+    auto describeType = [&](Type argType, const PrintOptions &PO) -> std::string {
+      std::string Result;
+      llvm::raw_string_ostream OS(Result);
+
+      OS << "'";
+      argType.print(OS, PO);
+      OS << "'";
+
+      if (auto *opaque = argType->getAs<OpaqueTypeArchetypeType>()) {
+        auto *decl = opaque->getDecl()->getNamingDecl();
+        OS << " (result type of '" << decl->getBaseName().userFacingName()
+           << "')";
+      } else if (auto archetype = argType->getAs<ArchetypeType>()) {
+        if (auto *GTD = getGenericTypeDecl(archetype))
+          OS << " (" << describeGenericType(GTD) << ")";
```

Comment:
@xedin I can't figure out how to get into this branch (which I moved here from lower down in the file where you originally added it in 5437622d2da556b23ab459452d2d4bcfb0dd0d13). If I remove this branch, all the tests pass. Can you help me figure out whether it's still needed, and if so how to exercise it?

Reply:
(This is now marked "Outdated" because I rearranged the code slightly, but the question is still relevant.)

Reply:
This is the way to trigger this branch (you'd need to use a parameter with generic type in the body of a function that declares it):

```
func other<U>(_: U, _: U) {}

struct S<T> {
  var myV: T
  func test<T>(t: T) {
     other(t,  myV)
  }
}
```